### ISSUE TITLE
Can't log in to ntfy

### ISSUE THREAD

**Issue by der-bender** [YunoHost-Apps/ntfy_ynh#49]
### Describe the bug

After upgrading to yunohost 12.0.6 I can't log in to ntfy. ntfy doesn't accept the previously working credentials. I changed the password via cli/ntfy.sh user, the change seems to be successful, but I still can't log in using that newly created password.

### Context

- Hardware: VPS bought online
- YunoHost version: 12.0.6
- I have access to my server: through SSH and webadmin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no, standard-install
- Using, or trying to install package version/branch: ntfy 
- If upgrading, current package version: 2.11.0~ynh1

### Steps to reproduce

- open ntfy's domain
- enter credentials

### Expected behavior

Getting logged in to ntfy. Instead, I get an error message stating "Login failed: Invalid username or password".

**Comment by l-x** [comment]
Thanks for the message, I actually have the same problem - but unfortunately no idea at the moment what the reason could be.

I have the assumption that it could be a problem with the shared sqlite library, since the login is no longer possible since the Yunhost update. There has been no change to either ntfy or ntfy-ynh.

**Comment by der-bender** [comment]
Tanks for your reply! Chatted about the issue yesterday via ynh support yesterday, nobody had an idea until now but the problem seems to be reproducable.

**Comment by l-x** [comment]
It doesn't even work with a complete reinstallation...

**Comment by der-bender** [comment]
I also tried that. Additionally ntfy is marked as broken in the webadmin interface. After declaring that you're taking the risk of breaking your system ntfy seems to install fine, but the problem persists.

**Comment by l-x** [comment]
> marked as broken in the webadmin

That's a useful hint, I did not know. Maybe it's time to release a new version... :D

**Comment by der-bender** [comment]
The shown breakage may be connected with problems shown here: https://ci-apps.yunohost.org/ci/apps/ntfy/
but I only guess. Unfortunately I'm not a programmer so I'm of little help, I'm afraid...

**Comment by l-x** [comment]
I'm going to publish a new app release, which is hopefully no more broken ^^. After that I have to look into the authentication issue, in case it's not just solved itself.

**Comment by der-bender** [comment]
Let me know if I can be of help.

**Comment by l-x** [comment]
> Let me know if I can be of help.

Thanks alot, you already helped by creating this issue ^^

**Comment by der-bender** [comment]
Whoa, man already a new release - thanks! Really appreciate your effort. Updated my old installation but still can't log in. Uninstalled and reinstalled an still can't log in. Am I doing something wrong or should I simply be patient?

**Comment by l-x** [comment]
> Am I doing something wrong or should I simply be patient?

No, you're doing nothing wrong. The problem still exists and I still have no clue about the reason. I guess you will have to be patient, sorry...

**Comment by der-bender** [comment]
Well, that's no problem at all! I really am thankful for all the work you're putting into this!

**Comment by l-x** [comment]
Apparently the nginx has stopped forwarding the authentication headers to the ntfy server. This is one step further, because at least I know the reason for the problem.

Stay tuned and sorry for the inconvenience!

**Comment by der-bender** [comment]
Well, no problem, thanks again for the effort you're putting into this
project!

**Comment by l-x** [comment]
Auth header gets stripped by ssowat.

https://forum.yunohost.org/t/yunohost-12-0-alpha-bookworm/28188/43

**Comment by l-x** [comment]
Upstream Issue: https://github.com/YunoHost/issues/issues/2422

A temporary workaround is to ssh to you server and execute

```sh
sudo yunohost tools shell -c "from yunohost.app import(app_setting, app_ssowatconf); app_setting('<app>','auth_header', 'basic-with-password'); app_setting('<app>', 'protect_against_basic_auth_spoofing', False); app_ssowatconf()"
```
in your terminal. Replace `<app>` with your apps name - which might be just `ntfy`.
in the shell.

**Comment by der-bender** [comment]
The Workaround did the trick, thanks!

**Comment by xmgz** [comment]
thank you, I will try to reinstall. I've running in circles for two days trying to solve what I thought it was my problem.

I will make a clean install and start again after messing things badly (I presume)

**Comment by MadMan247** [comment]
Hello. I just installed ntfy for the first time, and I would like to say that this is still very much an issue. I do not think that this issue should be marked as closed. Please consider reopening it.

**Comment by l-x** [comment]
@Tagadda @stepcellwolf @Quiwy @der-bender

You haven't heard from me in a while, but I think I've found a solution to that age-old login problem.

If you can and are willing to, please test the testing branch:

```sh
sudo yunohost app install https://github.com/YunoHost-Apps/ntfy_ynh/tree/testing
```

I'd really appreciate your feedback!

**Comment by l-x** [comment]
Fixed with f2544e2
